tests/unit/test_umm_var.py

```diff
@@ -758,3 +763,74 @@ def test_export_all_umm_var_to_json(self):
+        # And set the return_value of `mock_requests_put` to mock_response
+        mock_response.json.return_value = {'concept-id': 'FOO-EEDTEST'}
+        mock_requests_put.return_value = mock_response
+        umm_var_dict = {'Name': 'Test', 
+                        'MetadataSpecification':
+                            {'URL': 'https://foo.gov/umm/variable/v1.8.2',
+                            'Name': 'UMM-Var',
+                            'Version': '1.8.2'}}
+                
+        concept_id = 'C1256535511-EEDTEST'
+        cmr_env = CMR_UAT
+        token = 'foo'
+        headers_umm_var = {
+            'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
+            + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
+            'Authorization': f'Bearer {token}',
+            'Accept': 'application/json'}
+        
+        url_endpoint = (cmr_env.replace('search', 'ingest') + 'collections/'
+                    f'{concept_id}/variables/{umm_var_dict["Name"]}')
+        
+        response = publish_umm_var(concept_id, umm_var_dict, token, cmr_env)
+        
+        # Check if `publish_umm_var` was called once with expected parameters
+        mock_requests_put.assert_called_once_with(url_endpoint,
+                                                  json=umm_var_dict,
+                                                  headers=headers_umm_var,
+                                                  timeout=10)
+        
+        with self.subTest('Check response when the status code is 400'):
+            self.assertEqual(response.status_code, 400)
+            self.assertEqual(response.json(), {'concept-id': 'FOO-EEDTEST'})
+        
+        with self.subTest('Check response when the status code is 200'):
+            mock_response.status_code = 200
+            response = publish_umm_var(concept_id, umm_var_dict, token, cmr_env)
+            self.assertEqual(response, 'FOO-EEDTEST')
+
+
+    def test_publish_all_umm_var(self):
+        ''' Check if `publish_all_umm_var` returns expected number of
+            published UMM-Var entries.
+        '''
+        umm_var_dict = {'Test_1': {'Name': 'Test_1', 
+                        'MetadataSpecification':
+                            {'URL': 'https://foo.gov/umm/variable/v1.8.2',
+                            'Name': 'UMM-Var',
+                            'Version': '1.8.2'}},
+                        'Test_2': {'Name': 'Test_2', 
+                        'MetadataSpecification':
+                            {'URL': 'https://foo.gov/umm/variable/v1.8.2',
+                            'Name': 'UMM-Var',
+                            'Version': '1.8.2'}}}
+                
+        variable_ids = publish_all_umm_var('C1256535511-EEDTEST',
+                                           umm_var_dict,
+                                           'foo',
+                                           CMR_UAT)
+        self.assertEqual(len(variable_ids), 2)
```

Comment:
That's a great question and something that @owenlittlejohns and other folks have been talking about. We don't want to raise an error because raising an error would stop the whole ingest process if a single variable in a collection has a hundred variables. We want to  add it into the list and maybe have something that checks later for if they were any errors. But it is a good idea to add a test for checking if the list contains errors. I will add that.

Reply:
Agree we do not want to fail the "put" (ingest), but there is a good chance this will be updated/extended in future releases.